packages/@aws-cdk/core/lib/cfn-parse.ts

```diff
@@ -220,3 +308,7 @@ function specialCaseRefs(value: any): any {
     default: return undefined;
   }
 }
+
+function undefinedIfAllValuesAreEmpty(object: object): object | undefined {
```

Comment:
Terser name: `undefinedIfEmpty` should suffice.

Reply:
In my opinion, `undefinedIfEmpty` would mean "return undefined if this object has no keys", which is not what this does, so it would be quite a confusing name actually.